lib/src/tree.rs

```diff
@@ -309,87 +309,99 @@ impl<T> Diff<T> {
+                                    let before = it1.next().unwrap();
+                                    return Some((
+                                        before.name().clone(),
+                                        Some(before.value()),
+                                        None,
+                                    ));
+                                }
+                                Ordering::Greater => {
+                                    // entry added
+                                    let after = it2.next().unwrap();
+                                    return Some((after.name().clone(), None, Some(after.value())));
+                                }
+                                Ordering::Equal => {
+                                    // entry modified or clean
+                                    let before = it1.next().unwrap();
+                                    let after = it2.next().unwrap();
+                                    if before.value() != after.value() {
+                                        return Some((
+                                            before.name().clone(),
+                                            Some(before.value()),
+                                            Some(after.value()),
+                                        ));
+                                    }
+                                }
+                            }
+                        }
+                        (Some(_), None) => {
+                            // second iterator exhausted
+                            let before = it1.next().unwrap();
                             return Some((before.name().clone(), Some(before.value()), None));
                         }
-                        Ordering::Greater => {
-                            // entry added
-                            let after = self.it2.next().unwrap();
+                        (None, Some(_)) => {
+                            // first iterator exhausted
+                            let after = it2.next().unwrap();
                             return Some((after.name().clone(), None, Some(after.value())));
                         }
-                        Ordering::Equal => {
-                            // entry modified or clean
-                            let before = self.it1.next().unwrap();
-                            let after = self.it2.next().unwrap();
-                            if before.value() != after.value() {
-                                return Some((
-                                    before.name().clone(),
-                                    Some(before.value()),
-                                    Some(after.value()),
-                                ));
-                            }
+                        (None, None) => {
+                            // both iterators exhausted
+                            return None;
                         }
                     }
                 }
```

Comment:
It's hard to tell from the diff, but the only change inside the loop is that `self.it` became `it`